src/gtest/test_sideChain.cpp

```diff
@@ -0,0 +1,1196 @@
+    CAmount duplicatedAmount = 100;
+    extendTransaction(aTransaction, duplicatedScId, duplicatedAmount);
+
+    uint256 anotherScId = uint256S("1912");
+    CAmount anotherScAmount = 2;
+    extendTransaction(aTransaction, anotherScId, anotherScAmount);
+
+    //prerequisites
+    ASSERT_TRUE(firstScId == duplicatedScId)<<"Test requires second tx to be a duplicate";
+    ASSERT_TRUE(firstScId != anotherScId)<<"Test requires third tx to be a valid one";
+
+    //test
+    bool res = coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    //check
+    EXPECT_FALSE(res);
+    EXPECT_TRUE(coinViewCache.sidechainExists(firstScId))
+            << "First, valid sidechain creation txs should be cached";
+    EXPECT_FALSE(coinViewCache.sidechainExists(anotherScId))
+            << "third, valid sidechain creation txs is currently not cached";
+}
+
+TEST_F(SideChainTestSuite, ForwardTransfersToNonExistentScAreRejected) {
+    uint256 nonExistentId = uint256S("1492");
+    CAmount initialFwdAmount = 1987;
+    aTransaction = createFwdTransferTxWith(nonExistentId, initialFwdAmount);
+
+    //Prerequisite
+    ASSERT_FALSE(coinViewCache.sidechainExists(nonExistentId))
+        <<"Test requires target sidechain to be non-existent";
+
+    //test
+    bool res = coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    //check
+    EXPECT_FALSE(res);
+    EXPECT_FALSE(coinViewCache.sidechainExists(nonExistentId));
+}
+
+TEST_F(SideChainTestSuite, ForwardTransfersToExistentSCsAreRegistered) {
+    //insert the sidechain
+    uint256 newScId = uint256S("1492");
+    CAmount initialFwdAmount = 1953;
+    aTransaction = createSideChainTxWith(newScId, initialFwdAmount);
+
+    coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    //create forward transfer
+    CAmount anotherFwdAmount = 1987;
+    aTransaction = createFwdTransferTxWith(newScId, anotherFwdAmount);
+
+    //Prerequisite
+    ASSERT_TRUE(coinViewCache.sidechainExists(newScId))
+        <<"Test requires Sc to exist before attempting the forward transfer tx";
+
+    //test
+    bool res = coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    //check
+    EXPECT_TRUE(res);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////// Flush /////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(SideChainTestSuite, FlushAlignsPersistedTxsWithViewOnes) {
+    uint256 newScId = uint256S("a1b2");
+    CAmount initialFwdTxAmount = 1;
+    int scCreationHeight = 10;
+    aTransaction = createSideChainTxWith(newScId, initialFwdTxAmount);
+    coinViewCache.UpdateScInfo(aTransaction, aBlock, scCreationHeight);
+
+    //prerequisites
+    ASSERT_TRUE(sideChainManager.sidechainExists(newScId,&coinViewCache))
+        << "Test requires a tx to be ready to be persisted";
+
+    //test
+    bool res = coinViewCache.Flush();
```

Comment:
Could it be useful asserting also res? Otherwise it is not used

Reply:
Agree. Introduced check on res.